Teacher onsite webapp

willgearty opened the pull request:
A mobile-friendly interface for teachers to perform common functions that might be desired at a Splash program:

- Live schedule
- Google maps of campus (with directions to classrooms if implemented)
- Section details
- Section rosters (with attendance functionality to come in the future)
- A page for the Splash teacher survey and a separate page to see student survey responses (both of which could be overridden with google forms or something)

Fixes https://github.com/learning-unlimited/ESP-Website/issues/2671.

**Screenshots:**

Schedule:
![image](https://user-images.githubusercontent.com/7232514/73235966-3c20d900-4156-11ea-9e67-0cb4c422e74d.png)

Class Details (if you click the details link in the top navbar):
![image](https://user-images.githubusercontent.com/7232514/73236020-80ac7480-4156-11ea-8588-b93f2f1b0266.png)

Class Rosters (if you click the rosters link in the bottom navbar from the class details page):
![image](https://user-images.githubusercontent.com/7232514/73236808-42648480-4159-11ea-9fff-16b250cbed55.png)

Individual Section Details (if you click the list icon next to a section on the schedule):
![image](https://user-images.githubusercontent.com/7232514/71632464-b0f70680-2bdc-11ea-9fe4-6693ed35a43a.png)

Individual Section Roster (if you click the clipboard icon next to a section on the schedule OR click the roster link from the details page):
![image](https://user-images.githubusercontent.com/7232514/73236839-72ac2300-4159-11ea-93ba-60e52b830445.png)

Teacher Survey:
![image](https://user-images.githubusercontent.com/7232514/71632492-d5eb7980-2bdc-11ea-86a5-c46d7fabf355.png)

Student Survey Results:
![image](https://user-images.githubusercontent.com/7232514/71632509-e0a60e80-2bdc-11ea-8ea1-f6c2f976d2a5.png)

willgearty commented:
I think the remaining LGTM alerts are fine to ignore.

hwatheod commented on esp/templates/program/modules/teacheronsite/schedule.html:
```diff
@@ -0,0 +1,45 @@
+            <i>Classes for {{user.first_name}} {{user.last_name}} - ID: {{user.id}}</i>
+            {% if not checked_in %}
+                <p class="checkinnote">Note: Please make sure to check in before your first class today.</p>
+            {% endif %}
```
I suggest making this text customizable with a tag, so that you don't have to do a template override just to change this message.

willgearty commented:
I can't think of anything else to implement at the moment for the teacher webapp. Please let me know if you have any comments or suggestions!

kkbrum commented:
If there are no coteachers, it probably shouldn't have the email link:
![image](https://user-images.githubusercontent.com/25392075/75116413-1a702000-5636-11ea-9f99-403d4f1907ff.png)

Also looks like the mailto link might be wrong? Shouldnt it have the class code at the start? (the students version have it)
![image](https://user-images.githubusercontent.com/25392075/75116445-79ce3000-5636-11ea-9ee8-e43bfede503f.png)


I personally think the "submit attendance" button should be bigger than the "scan barcodes with your device" button, just so people know they have to submit for it to be saved.

When I click surveys, I get the following message. Could we check whether there are surveys and if not, remove that header?
![image](https://user-images.githubusercontent.com/25392075/75116538-70919300-5637-11ea-9454-4a715176e059.png)

kkbrum commented:
It looks like a deadline is required for webapp, but doesn't exist as an option on the deadlines page.

kkbrum commented:
Actually though, when you were the sole teacher of your class (in your screenshots), it shows you as the sole coteacher. I dont know why I didn't show up on my list of coteachers...

willgearty commented:
Ah, I have multiple accounts on my dev server with the same name (so that class actually has 2 teachers)

willgearty commented:
> Actually though, when you were the sole teacher of your class (in your screenshots), it shows you as the sole coteacher. I dont know why I didn't show up on my list of coteachers...

Ah, I have multiple accounts on my dev server with the same name (so that class actually has 2 teachers)

willgearty commented:
Thanks for the review @kkbrum. I believe I've addressed all of your comments, and I fixed the merge conflict with the changes from the attendance module.

kkbrum commented:
Maybe I'm doing something wrong, but I think I closed the webapp deadline and yet when I morph into a non-admin teacher, I still see the webapp, not an error page. My deadlines are as follows:

![image](https://user-images.githubusercontent.com/25392075/75733348-13df4980-5cc3-11ea-8f36-2ffb3a72dc9e.png)

![image](https://user-images.githubusercontent.com/25392075/75733355-193c9400-5cc3-11ea-924b-16425d95564e.png)


Everything else looks good.
